### PR TITLE
Fix available storage decrease issue when a vm allocated on vm host

### PR DESCRIPTION
**Pull request by enescakir**
### Fix storage_volumes value in the stack for testing
`storage_size_gib` should be `size_gib` in the frame. The current usage in the tests is incorrect.
https://github.com/ubicloud/ubicloud/blob/d96815d8fafb12ef369e1c7cc7b0f2cd0c713edb/lib/validation.rb#L64C5-L64C5

We only use `storage_volumes` from the stack at the `start` label, after which we clear it. Other labels should use `VmStorageVolume` records. Consequently, I moved this general 'allow' block to the `.assemble` block only.

### Fix available storage decrease issue when a vm allocated on vm host
I noticed unusually high values for `available_storage_gib` on our VM hosts

Recently, we moved the VmStorageVolume creation from before VM host selection to after, as seen in commit d9b7d58. The `vm.storage_size_gib` relies on these models to compute total used storage. Therefore, the line `available_storage_gib: Sequel[:available_storage_gib] - vm.storage_size_gib` has no effect, as the VM doesn't yet have storage volume records.

To address this issue, I changed it to return the storage size from the strand's stack if the vm doesn't have storage volumes.

We also need to correct these values in production after deploying this fix.

### CODE REVIEW

**Review by furkansahin — APPROVED** [review]
Wow, that's a good catch 🤯

**Comment by fdr** [comment]
This is pretty brutal, it doesn't seem like a viable long-term fix, to go fishing around in stack frames. Is this intended to be expedient or transitory?

**Comment by enescakir** [comment]
> This is pretty brutal, it doesn't seem like a viable long-term fix, to go fishing around in stack frames. Is this intended to be expedient or transitory?

I didn't think this one as transitory. @pykello might have better solution?

File: model/vm.rb [hunk 1]
```diff
@@ -149,7 +149,11 @@ def inhost_name
   end
 
   def storage_size_gib
-    vm_storage_volumes.map { _1.size_gib }.sum
+    if vm_storage_volumes.empty?
+      strand.stack.first.fetch("storage_volumes", []).map { _1["size_gib"] }.sum
```
[the remark this answers is not in the document]
Comment (enescakir) [reply model/vm.rb]: I converted to this version

**Review by pykello — APPROVED** [review]

**Review by fdr — APPROVED** [review]
I think @pykello got it right, by moving it to the nexus it makes more sense, rather than in a model where anything calling the symbol without the right level of locking will be getting something rather strange...and we have no use for something quite that contingent yet.

**Comment by enescakir** [comment]
The tests started to fail after rebase in this branch. I added another commit to fix specs first. Can you take another look? @fdr @pykello @furkansahin 

### Fix storage_volumes value in the stack for testing

`storage_size_gib` should be `size_gib` in the frame. The current usage in the tests is incorrect.
https://github.com/ubicloud/ubicloud/blob/d96815d8fafb12ef369e1c7cc7b0f2cd0c713edb/lib/validation.rb#L64C5-L64C5

We only use `storage_volumes` from the stack at the `start` label, after which we clear it. Other labels should use VmStorageVolume records. Consequently, I moved this general 'allow' block to the `.assemble` block only.

**Review by pykello — APPROVED** [review]

**Pull request merged**